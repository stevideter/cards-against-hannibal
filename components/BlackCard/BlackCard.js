import styles from './BlackCard.module.css'

const BlackCard = () => (
    <div className={styles.card}>
        <p>
        This is my ____.
   
        </p> </div>

  );
  
  export default BlackCard;