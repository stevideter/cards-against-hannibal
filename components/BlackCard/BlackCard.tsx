import styled from 'styled-components';

const StyledBlackCard = styled.div`
    background: black;
    color: white;
    width: 150px;
    height: 300px;
    padding: 10px;
    font-size: xx-large;
    font-weight: bolder;
    border-radius: 25px;
    border: 2px solid black;
    position: relative;
`;
const StyledP = styled.p`
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
`;
interface BlackCardProps {
    text: string;
}

const BlackCard = (props: BlackCardProps): JSX.Element => (
    <StyledBlackCard>
        <StyledP>{props.text}</StyledP>
    </StyledBlackCard>
);

export default BlackCard;
