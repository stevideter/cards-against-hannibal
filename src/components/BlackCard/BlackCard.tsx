import styled from 'styled-components';

const StyledBlackCard = styled.div`
    background: black;
    box-shadow: rgb(102, 102, 102) 0px 2px 2px 0px;
    box-sizing: border-box;
    color: white;
    width: 10em;
    height: 10em;
    padding: 10px;
    border-radius: 10px;
    border: 2px solid black;
    position: relative;
    margin: 10px;
`;
const StyledP = styled.p`
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    font-size: 1.3em;
    font-family: MuseoSans, Arial, Helvetica, sans-serif;
    text-size-adjust: 100%;
`;
interface BlackCardProps {
    id: string;
    text: string;
}

const BlackCard: React.FunctionComponent<BlackCardProps> = (
    props: BlackCardProps
) => (
    <StyledBlackCard id={props.id}>
        <StyledP>{props.text}</StyledP>
    </StyledBlackCard>
);

export default BlackCard;
