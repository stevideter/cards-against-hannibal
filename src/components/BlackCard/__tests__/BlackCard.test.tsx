import BlackCard from '../BlackCard';
import renderer from 'react-test-renderer';

describe('BlackCard', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<BlackCard text="text" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
