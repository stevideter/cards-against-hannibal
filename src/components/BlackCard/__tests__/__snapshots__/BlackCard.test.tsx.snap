// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlackCard renders correctly 1`] = `
<div
  className="sc-AxjAm iSiQYW"
>
  <p
    className="sc-AxirZ kqXkCR"
  >
    text
  </p>
</div>
`;
